AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py

```diff
@@ -0,0 +1,75 @@
+from flojoy import flojoy, run_in_venv, Image, DataFrame
+
+
+@flojoy
+@run_in_venv(
+    pip_dependencies=[
+        "transformers",
+        "torch",
+        "Pillow",
+        "numpy",
+        "pandas",
+    ],
+    verbose=True,
+)
+def HUGGING_FACE_PIPELINE(
+    default: Image,
+    model: str = "google/vit-base-patch16-224",
+    revision: str = "main",
+) -> DataFrame:
+    """Hugging Face Pipeline for Image Classification.
+
+    Parameters
+    ----------
+    default: Image
+        The input image to be classified. The image must be a PIL.Image object wrapped in a flojoy Image object.
+    model: str
+        The model to be used for classification.
+        If not specified, Vision Transformers (i.e. `google/vit-base-patch16-224`) are used.
+        For more information about Vision Transformers, see: https://huggingface.co/google/vit-base-patch16-224
+        For a complete list of models see: https://huggingface.co/models?pipeline_tag=image-classification
+    revision: str
+        The revision of the model to be used for classification.
+        If not specified, main is `used`. For instance see: https://huggingface.co/google/vit-base-patch16-224/commits/main
+
+    Returns
+    -------
+    DataFrame:
+        A DataFrame containing as columns the `label` classification label and `score`, its confidence score.
+        All scores are between 0 and 1 and sum to 1.
+    """
+
+    import os
+    import numpy as np
+    import pandas as pd
+    import PIL.Image as PILImage
+    from flojoy.utils import FLOJOY_CACHE_DIR
+    from typing import List, Dict
+
+    # Setting transformers cache directory to flojoy cache directory before importing transformers
+    # not to pollute the user's cache directory.
+    os.environ["TRANSFORMERS_CACHE"] = os.path.join(FLOJOY_CACHE_DIR, "transformers")
```

Comment:
We probably might want to make this systematic.

Reply:
+1

We previously wrapped the `huggingface-hub` library within `flojoy` though it's simpler to just ensure `os.environ["HF_HOME"]` is set whenever flojoy is imported

Reply:
Can we add it to Flojoy's processes' environment?

Reply:
https://github.com/flojoy-ai/python/pull/97 Incoming 👍